go-controller/pkg/kubevirt/router.go

```diff
@@ -0,0 +1,300 @@
+	}
+
+	skipPointToPointRouting, err := nodeContainsPodSubnet(watchFactory, pod.Spec.NodeName, podAnnotation, nadName)
+	if err != nil {
+		return err
+	}
+
+	if skipPointToPointRouting {
+		// Point to point routing is no longer needed if vm
+		// is running at the node with VM's subnet
+		if err := DeleteRoutingForMigratedPod(nbClient, pod); err != nil {
+			return err
+		}
+		return nil
+	} else {
+		// Since we are at remote zone we should not have local zone point to
+		// to point routing
+		if err := DeleteRoutingForMigratedPodWithZone(nbClient, pod, OvnLocalZone); err != nil {
+			return err
+		}
+	}
+
+	node, err := watchFactory.GetNode(pod.Spec.NodeName)
+	if err != nil {
+		return err
+	}
+	transitSwitchPortAddrs, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil {
+		return err
+	}
+	for _, podIP := range podAnnotation.IPs {
+		ipFamily := utilnet.IPFamilyOfCIDR(podIP)
+		transitSwitchPortAddr, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, transitSwitchPortAddrs)
+		if err != nil {
+			return err
+		}
+		route := nbdb.LogicalRouterStaticRoute{
+			IPPrefix: podIP.IP.String(),
+			Nexthop:  transitSwitchPortAddr.IP.String(),
+			Policy:   &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			ExternalIDs: map[string]string{
+				OvnZoneExternalIDKey:         OvnRemoteZone,
+				VirtualMachineExternalIDsKey: pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+				NamespaceExternalIDsKey:      pod.Namespace,
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(nbClient, types.OVNClusterRouter, &route, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == route.IPPrefix && item.Nexthop == route.Nexthop && item.Policy != nil && *item.Policy == *route.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route at remote zone: %v", err)
+		}
+	}
+	return nil
+}
+
+func virtualMachineReady(watchFactory *factory.WatchFactory, pod *corev1.Pod) (bool, error) {
+	isMigratedSourcePodStale, err := IsMigratedSourcePodStale(watchFactory, pod)
+	if err != nil {
+		return false, err
+	}
+	if util.PodWantsHostNetwork(pod) || !IsPodLiveMigratable(pod) || isMigratedSourcePodStale {
+		return false, nil
+	}
+
+	// When a virtual machine start up this
+	// label is the signal from KubeVirt to notify that the VM is
+	// ready to receive traffic.
+	targetNode := pod.Labels[kubevirtv1.NodeNameLabel]
+
+	// This annotation only appears on live migration scenarios and it signals
+	// that target VM pod is ready to receive traffic so we can route
+	// taffic to it.
+	targetReadyTimestamp := pod.Annotations[kubevirtv1.MigrationTargetReadyTimestamp]
+
+	// VM is ready to receive traffic
+	return targetNode == pod.Spec.NodeName || targetReadyTimestamp != "", nil
+}
+
+func nodeContainsPodSubnet(watchFactory *factory.WatchFactory, nodeName string, podAnnotation *util.PodAnnotation, nadName string) (bool, error) {
```

Comment:
For `EnsureRemoteZonePodAddressesToNodeRoute` to remove the transit switch port static route when the VM migrate back to node owning the subnet we checking the logical swtich manager will not work since it will never has the switch there (we are at the remote zone).